pkg/build/metrics/prometheus/metrics.go

```diff
@@ -77,72 +77,79 @@ func (bc *buildCollector) Collect(ch chan<- prometheus.Metric) {
 		return
 	}
 
-	// since we do not collect terminal build metrics on a per build basis, collectBuild will return counts
-	// to be added to the total amount posted to prometheus
-	var cancelledCount, completeCount, errorCount, newCount, pendingCount, runningCount int
-	reasons := map[string]int{}
+	// collectBuild will return counts for the build's phase/reason tuple,
+	// and counts for these tuples be added to the total amount posted to prometheus
+	counts := map[string]map[string]int{
+		cancelledPhase: map[string]int{},
+		completePhase:  map[string]int{},
+		errorPhase:     map[string]int{},
+		failedPhase:    map[string]int{},
+		newPhase:       map[string]int{},
+		pendingPhase:   map[string]int{},
+		runningPhase:   map[string]int{},
+	}
 	for _, b := range result {
-		cc, cp, ec, nc, pc, rc, r := bc.collectBuild(ch, b)
-		for key, value := range r {
-			reasons[key] = reasons[key] + value
+		r := bc.collectBuild(ch, b)
+		for phase, reasonMap := range r {
+			currentPhaseMap := counts[phase]
+			for reason, count := range reasonMap {
+				if _, ok := currentPhaseMap[reason]; !ok {
+					currentPhaseMap[reason] = 0
```

Comment:
shouldn't be necessary - reading the map when the key doesn't exist will return the zero value.

Reply:
I had wondered about that given how it was but couldn't remember and then forgot to look and potentially change 

Thanks for the heads up - will change in next push